File: android/src/main/kotlin/com/situm/flutter/ar/situm_ar/ARMethodCallSender.kt
package com.situm.flutter.ar.situm_ar

import io.flutter.plugin.common.MethodChannel

/**
 * MethodChannel wrapper/adapter. Use this class to communicate with the Dart side.
 */
class ARMethodCallSender(
    private val methodChannel: MethodChannel
) {
    fun sendArGoneRequired() {
        methodChannel.invokeMethod("ArGoneRequired", mapOf("reason" to "lifecycle_stop" ))
    }
}

File: ios/Classes/ARMethodCallSender.swift
import Foundation
import Flutter

/**
 * MethodChannel wrapper/adapter. Use this class to communicate with the Dart side.
 */
class ARMethodCallSender {
    
    private let methodChannel: FlutterMethodChannel
    
    init(methodChannel: FlutterMethodChannel) {
        self.methodChannel = methodChannel
    }
    
    // Method to send messages to Dart when or AR must be stopped
    func sendArGoneRequired() {
        let arguments: [String: Any] = ["reason": "lifecycle_stop"]
        methodChannel.invokeMethod("ArGoneRequired", arguments: arguments)
    }
}
